### PR TITLE
Fix #214

### LINKED ISSUE #214: Misleading error message for multiple defined function
Consider this code:

    def f() -> int: return 1
    def f() -> int: return 2

The type checker gives this type error:

    t.py, line 2: Name 'f' already defined (overload variants 
    must be next to each other)

This is misleading, since the functions are next to each other and this is not a case of overloading but redefinition.

### PR DESCRIPTION
**Pull request by thomasballinger**
A few questions:

* Is it OK that error message refers to a line number, and uses `self.globals[n].node.line` to get it? Will that always exist?
* Is it OK that I removed a test? It should be added again once things like https://github.com/python/mypy/issues/409 are fixed.

### CODE REVIEW

File: mypy/test/data/check-functions.test [hunk 1]
```diff
@@ -1006,6 +1006,9 @@ tmp/a.py:5: error: "str" not callable
 -- Conditional function definition
 -- -------------------------------
 
+[case testUnconditionalRedefinitionOfUnconditionalFunction]
```
Comment (gvanrossum) [inline mypy/test/data/check-functions.test]: Can you find a better place for this test? It looks kind of out of place here.

File: mypy/test/data/check-overloading.test [hunk 2]
```diff
@@ -91,12 +91,6 @@ class A:
   def f(self, x: 'B') -> 'B': pass
 class B: pass
 
-[case testNonConsecutiveOverloads]
```
Comment (gvanrossum) [inline mypy/test/data/check-overloading.test]: I think what happened here is that long ago mypy allowed function overloading without `@overload`. This test was never updated and neither was the error message! (So it's fine to delete. :-)

File: mypy/semanal.py [hunk 3]
```diff
@@ -2210,8 +2210,8 @@ def check_no_global(self, n: str, ctx: Context,
                         is_func: bool = False) -> None:
         if n in self.globals:
             if is_func and isinstance(self.globals[n].node, FuncDef):
-                self.fail(("Name '{}' already defined (overload variants "
-                           "must be next to each other)").format(n), ctx)
+                self.fail(("Name '{}' already defined on line "
+                           "{}").format(n, self.globals[n].node.line), ctx)
```
Comment (gvanrossum) [inline mypy/semanal.py]: Occasionally a node has no line number, and then `line` is `-1`. Can you check for that?

You could also consider emitting the line number of the previous definition as a separate message with `severity='note'`. That makes it possible for e.g. Emacs to jump to that line separately.

Comment (thomasballinger) [inline mypy/semanal.py]: I removed this for now, instead using the regular code path for repeat definition. Referring to the previous line sounds like something that every repeat definition could benefit from.

**Comment by thomasballinger** [comment]
I've moved the test and removed the line number specific error message, instead using the default `name_already_defined` method.

**Comment by gvanrossum** [comment]
Great! So glad to see some code deleted. :-)

**Pull request merged**